composer/2022_airflow_summit/data_analytics_process_expansion_test.py

```diff
@@ -0,0 +1,233 @@
+BUCKET_BLOB = "data_analytics_process_expansion.py"
+TEST_CSV_FILE = "test_data_expansion.csv"
+
+BQ_CLIENT = bigquery.Client(project=PROJECT_ID)
+
+# BigQuery configs
+BQ_DESTINATION_DATASET_NAME = f"expansion_project_test_{TEST_ID}".replace("-", "_")
+BQ_DESTINATION_TABLE_NAME = "ghcnd_stations_joined"
+BQ_DESTINATION_TABLE_ID = (
+    f"{PROJECT_ID}.{BQ_DESTINATION_DATASET_NAME}.{BQ_DESTINATION_TABLE_NAME}"
+)
+BQ_NORMALIZED_TABLE_NAME = "ghcnd_stations_normalized"
+BQ_PRCP_MEAN_TABLE_NAME = "ghcnd_stations_prcp_mean"
+BQ_SNOW_MEAN_TABLE_NAME = "ghcnd_stations_prcp_mean"
+BQ_PHX_PRCP_TABLE_NAME = "phx_annual_prcp"
+BQ_PHX_SNOW_TABLE_NAME = "phx_annual_snow"
+
+PYSPARK_JAR = "gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar"
+PROCESSING_PYTHON_FILE = f"gs://{BUCKET_NAME}/{BUCKET_BLOB}"
+
+
+@pytest.fixture(scope="module")
+def test_dataproc_batch():
+
+    BATCH_ID = f"summit-dag-expansion-test-{TEST_ID}"  # Dataproc serverless only allows lowercase characters
+    BATCH_CONFIG = {
+        "pyspark_batch": {
+            "jar_file_uris": [PYSPARK_JAR],
+            "main_python_file_uri": PROCESSING_PYTHON_FILE,
+            "args": [
+                BUCKET_NAME,
+                f"{BQ_DESTINATION_DATASET_NAME}.{BQ_DESTINATION_TABLE_NAME}",
+                f"{BQ_DESTINATION_DATASET_NAME}.{BQ_NORMALIZED_TABLE_NAME}",
+                f"{BQ_DESTINATION_DATASET_NAME}.{BQ_PRCP_MEAN_TABLE_NAME}",
+                f"{BQ_DESTINATION_DATASET_NAME}.{BQ_SNOW_MEAN_TABLE_NAME}",
+                f"{BQ_DESTINATION_DATASET_NAME}.{BQ_PHX_PRCP_TABLE_NAME}",
+                f"{BQ_DESTINATION_DATASET_NAME}.{BQ_PHX_SNOW_TABLE_NAME}",
+            ],
+        }
+    }
+
+    yield (BATCH_ID, BATCH_CONFIG)
+    dataproc_client = dataproc.BatchControllerClient(
+        client_options={
+            "api_endpoint": f"{DATAPROC_REGION}-dataproc.googleapis.com:443"
+        }
+    )
+    request = dataproc.DeleteBatchRequest(
+        name=f"projects/{PROJECT_ID}/locations/{DATAPROC_REGION}/batches/{BATCH_ID}"
+    )
+
+    # Declare variable outside of try/except so it can be printed in the exception
+    response = None
+    try:
+        # Make the request
+        response = dataproc_client.delete_batch(request=request)
+    except NotFound:
+        # There will only be a response if the deletion fails
+        # otherwise response will be None
+        print(response)
+
+
+@pytest.fixture(scope="module")
+def test_bucket():
+    # Create GCS Bucket
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(BUCKET_NAME)
+    print(os.listdir())
+
+    # Upload dag processing file
+    blob = bucket.blob(BUCKET_BLOB)
+    blob.upload_from_filename(BUCKET_BLOB)
+
+    # Upload test csv file
+    blob2 = bucket.blob(TEST_CSV_FILE)
+    blob2.upload_from_filename(TEST_CSV_FILE)
+
+    yield
+
+    # Delete GCS bucket
+    bucket = storage_client.get_bucket(BUCKET_NAME)
+    bucket.delete(force=True)
+
```

Comment:
Print statements here after could help.

Reply:
What would you like to be printed? Just confirmation that things were uploaded/deleted?

Reply:
Yeah for debugging purposes, but fine to leave as is if you think it's fine to just see that the test passes/fails.

Reply:
leaving as is to match other test